### PR TITLE
R snippet callbacks

### PR DESCRIPTION
**Pull request by AnnMarieW**

### CODE REVIEW

**Comment by tcbegley** [comment]
Thanks @AnnMarieW!

Does this supersede #596?

File: docs/components_page/components/__tests__/test_toast.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    dashjl.start_server(jl_app)
+    check_toast_icon_callbacks(dashjl)
+
+
+def check_toast_icon_callbacks(runner):
+    wait.until(
+        lambda: len(runner.find_elements("#simple-toast")) > 0,
+        timeout=2,
+    )
+    runner.find_element(".close").click()
```
Comment (tcbegley) [inline docs/components_page/components/__tests__/test_toast.py:58]: This is a weird one, but it looks like for some reason if the close button is clicked too early here the Julia test fails. Can you add

```diff
+ time.sleep(2)
runner.find_element(".close").click()
```

then the tests should pass

File: docs/components_page/components/__tests__/test_collapse.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+def check_collapse_multiple_callbacks(runner):
+
+    runner.find_element("#left").click()
+    wait.until(
+        lambda: runner.find_element("#left-collapse").get_attribute("class")
+        == "collapse",
+        timeout=2,
+    )
+
```
Comment (tcbegley) [inline docs/components_page/components/__tests__/test_collapse.py:60]: Similarly to toast example, can you add a `time.sleep(2)` here, I think it helps make the test a bit more robust.

File: docs/components_page/components/__tests__/test_collapse.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    runner.find_element("#right").click()
+    wait.until(
+        lambda: runner.find_element("#right-collapse").get_attribute("class")
+        == "collapse",
+        timeout=2,
+    )
+
```
Comment (tcbegley) [inline docs/components_page/components/__tests__/test_collapse.py:67]: Ditto

**Review by tcbegley — CHANGES_REQUESTED** [review]
Thanks @AnnMarieW!

I have left some comments, I believe with the suggested changes all tests should pass reliably 🤞

File: docs/components_page/components/__tests__/test_collapse.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+def test_r_collapse_accordion(dashr):
+    r_app = load_r_app((HERE.parent / "collapse" / "accordion.R"), "accordion")
+    dashr.start_server(r_app)
+    check_collapse_accordion_callbacks(dashr)
+
+
+def test_jl_collapse_accordion(dashjl):
+    jl_app = load_jl_app(
+        (HERE.parent / "collapse" / "accordion.jl"), "collapses"
```
[the remark this answers is not in the document]
Comment (AnnMarieW) [reply docs/components_page/components/__tests__/test_collapse.py]: :facepalm:

**Comment by AnnMarieW** [comment]
I incorporated all your suggestions with the latest commit :bowing_woman:

**Pull request merged**